components/gateway/internal/auditlog/service.go

```diff
@@ -0,0 +1,191 @@
+package auditlog
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/kyma-incubator/compass/components/gateway/internal/auditlog/model"
+	"github.com/kyma-incubator/compass/components/gateway/pkg/proxy"
+	"github.com/pkg/errors"
+)
+
+type AuditlogMessage struct {
+	Request  string
+	Response string
+	proxy.Claims
+}
+
+type AuditlogSource struct {
+	logsChannel chan AuditlogMessage
+}
+
+func NewAuditlogSink(logsChannel chan AuditlogMessage) *AuditlogSource {
+	return &AuditlogSource{
+		logsChannel: logsChannel,
+	}
+}
+
+func (sink *AuditlogSource) Log(request, response string, claims proxy.Claims) error {
+	msg := AuditlogMessage{
+		Request:  request,
+		Response: response,
+		Claims:   claims,
+	}
+	sink.logsChannel <- msg
+	return nil
+}
+
+type DummyAuditlog struct {
+}
+
+func (sink *DummyAuditlog) Log(request, response string, claims proxy.Claims) error {
+	return nil
+}
+
+//go:generate mockery -name=AuditlogClient -output=automock -outpkg=automock -case=underscore
+type AuditlogClient interface {
+	LogConfigurationChange(change model.ConfigurationChange) error
+	LogSecurityEvent(event model.SecurityEvent) error
+}
+
+type AuditlogService struct {
+	client AuditlogClient
+}
+
+func NewService(client AuditlogClient) *AuditlogService {
+	return &AuditlogService{client: client}
+}
+
+func (svc *AuditlogService) Log(request, response string, claims proxy.Claims) error {
+	graphqlResponse, err := svc.parseResponse(response)
+	if err != nil {
+		return errors.Wrap(err, "while parsing response")
+	}
+
+	if len(graphqlResponse.Errors) == 0 {
+		log := svc.createConfigChangeLog(claims, request)
+		log.Attributes = append(log.Attributes, model.Attribute{
+			Name: "response",
+			Old:  "",
+			New:  "success",
+		})
+
+		err = svc.client.LogConfigurationChange(log)
+		return errors.Wrap(err, "while sending to auditlog")
+	}
+
+	if idx := svc.isInsufficientScopeError(graphqlResponse.Errors); idx != -1 {
+		graphqErr := graphqlResponse.Errors[idx]
+		data, err := json.Marshal(&graphqErr)
+		if err != nil {
+			return errors.Wrap(err, "while marshalling graphql err")
+		}
+
+		err = svc.client.LogSecurityEvent(model.SecurityEvent{
+			User: "proxy",
+			Data: string(data),
+		})
+		return errors.Wrap(err, "while sending secuity event to auditlog")
+	}
+
+	isReadErr, err := isReadError(graphqlResponse, request)
+	if err != nil {
+		return errors.Wrap(err, "while checking if error is read error")
+	}
+
+	if isReadErr {
```

Comment:
I think this can be refactored and simplified. What about:
```go
log := svc.createConfigChangeLog(claims, request)
if isReadErr {
  log.Attributes = append(log.Attributes, model.Attribute{
    Name: "response",
    Old:  "",
    New:  "success",
  })
} else {
  log.Attributes = append(log.Attributes, model.Attribute{
    Name: "response",
    Old:  "",
    New:  response,
  })
}
err = svc.client.LogConfigurationChange(log)
return errors.Wrap(err, "while sending configuration change")
```